{
  if (r instanceof FutureTask) {
    try {
      ((FutureTask)r).get();
    }
 catch (    InterruptedException e) {
      throw new AssertionError(e);
    }
catch (    ExecutionException e) {
      logger_.error("Error in executor futuretask",e);
    }
  }
}
